packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -24,6 +36,146 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
+      // secret -- secret concept does not correlate to SDK socket event
+      ownedBy: SDK_EVENT_OWNER.CREATOR,
+      status: SDK_EVENT_STATUS.ACTIVE,
+      created: new Date().toISOString(),
+      data: {}
+    };
+    const that = this;
+
+
+    that.spark.people.get('me')
+      .then((person) => {
+        that.eventEnvelope.createdBy = person.id;
+        that.eventEnvelope.orgId = person.orgId;
+      }).catch((e) => {
+        console.error(`Unable to get person info for memberships \
+          event envelope ${e.message}`);
+      });
+
+    // Register to listen to events
+    return this.spark.internal.mercury.connect()
+      .then(() => this.listenTo(this.spark.internal.mercury,
+        WEBEX_EVENT.TEAMS_ACTIVITY,
+        (event) => this.onWebexApiEvent(event)));
+  },
+
+  /**
+   * Trigger a membership related events.
+   * @param {Object} event
+   * @returns {undefined}
+   */
+  onWebexApiEvent(event) {
+    const {activity} = event.data;
+
+    /* eslint-disable no-case-declarations */
+    switch (activity.verb) {
+      case API_ACTIVITY_VERB.ADD:
+        const membershipCreatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.CREATED);
+
+        debug(`membership "created" payload: \
+          ${JSON.stringify(membershipCreatedEventData)}`);
+        this.trigger(SDK_EVENT.CREATED, membershipCreatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.LEAVE:
+        const membershipDeletedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.DELETED);
+
+        debug(`membership "deleted" payload: \
+          ${JSON.stringify(membershipDeletedEventData)}`);
+        this.trigger(SDK_EVENT.DELETED, membershipDeletedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ADD_MODERATOR:
+      case API_ACTIVITY_VERB.REMOVE_MODERATOR:
+        const membershipUpdatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.UPDATED);
+
+        debug(`membership "updated" payload: \
+          ${JSON.stringify(membershipUpdatedEventData)}`);
+        this.trigger(SDK_EVENT.UPDATED, membershipUpdatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ACKNOWLEDGE:
+        const membershipSeenEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.SEEN);
+
+        debug(`membership "updated" payload: \
+          ${JSON.stringify(membershipSeenEventData)}`);
+        this.trigger(SDK_EVENT.SEEN, membershipSeenEventData);
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs the data object for an event on the memberships resource,
+   * adhering to Hydra's Webehook data structure memberships.
+   * @param {Object} activity from mercury
+   * @param {Object} event type of "webhook" event
+   * @returns {Object} constructed event
+   */
+  getMembershipEvent(activity, event) {
+    const sdkEvent = JSON.parse(JSON.stringify(this.eventEnvelope));
```

Comment:
Is this just a way to clone the data? You might also consider clone or clone deep from lodash if the sdk includes it.

Reply:
Yes, happy to use another preferred method for style or efficiency.  Can you recommend an approach?

Reply:
`const sdkEvent = Object.assign({}, this.eventEnvelope);` will create a new object for you.

Reply:
Thanks @adamweeks I tried that but when I use Object.assign the eventEnvelope gets updated.   Here's an example from a node terminal:
`
> sdkEnvelope = {foo: 'bar', data: {}} 
{ foo: 'bar', data: {} } 
> sdkEvent = Object.assign({}, sdkEnvelope) 
{ foo: 'bar', data: {} } 
> sdkEvent.data.key = "value1" 
'value1' 
> sdkEvent 
{ foo: 'bar', data: { key: 'value1' } } 
> sdkEnvelope 
{ foo: 'bar', data: { key: 'value1' } }
`
This doesn't happen when I use the JSON.stringify approach.   Is this bad/inefficient?

Reply:
Ended up resolving this with lodash cloneDeep.